spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/NGramTransformTest.java

```diff
@@ -0,0 +1,246 @@
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test for NGramTransform plugin.
+ */
+public class NGramTransformTest extends HydratorTestBase {
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  protected static final ArtifactId DATAPIPELINE_ARTIFACT_ID =
+    NamespaceId.DEFAULT.artifact("data-pipeline", "3.5.0");
+  protected static final ArtifactSummary DATAPIPELINE_ARTIFACT = new ArtifactSummary("data-pipeline", "3.5.0");
+
+  private static final String MULTIPLE_FIELDS_DATASET = "MultipleFields";
+  private static final String SINGLE_FIELD_DATASET = "SingleField";
+  private static final String APP_NAME = "NGramTransformTest";
+  private static final String OUTPUT_FIELD = "ngrams";
+  private static final String FIELD_TO_BE_TRANSFORMED = "tokens";
+  private static final String ADDITIONAL_FIELD = "name";
+  private static final String[] TOKENS_DATASET1 = {"hi", "i", "am", "cdap"};
+  private static final String[] TOKENS_DATASET2 = {"how", "are", "you", "cdap"};
+  private static final String[] TOKENS_DATASET3 = {"hi", "i"};
+
+  private static final Schema SOURCE_SCHEMA_SINGLE = Schema.recordOf("sourceRecord",
+                                                                     Schema.Field.of(FIELD_TO_BE_TRANSFORMED, Schema.arrayOf(Schema.of(Schema.Type.STRING)))
+  );
+
+  private static final Schema SOURCE_SCHEMA_SINGLE_NEGATIVE_TEST = Schema.recordOf("sourceRecord",
+                                                                                   Schema.Field.of(FIELD_TO_BE_TRANSFORMED, Schema.of(Schema.Type.STRING))
+  );
+  private static final Schema SOURCE_SCHEMA_MULTIPLE = Schema.recordOf("sourceRecord",
+                                                                       Schema.Field.of(ADDITIONAL_FIELD, Schema.of(Schema.Type.STRING)),
+                                                                       Schema.Field.of(FIELD_TO_BE_TRANSFORMED, Schema.arrayOf(Schema.of(Schema.Type.STRING)))
+  );
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    // add the artifact for etl batch app
+    setupBatchArtifacts(DATAPIPELINE_ARTIFACT_ID, DataPipelineApp.class);
+    // add artifact for spark plugins
+    Set<ArtifactRange> parents = ImmutableSet.of(
+      new ArtifactRange(NamespaceId.DEFAULT.toId(), DATAPIPELINE_ARTIFACT_ID.getArtifact(),
+                        new ArtifactVersion(DATAPIPELINE_ARTIFACT_ID.getVersion()), true,
+                        new ArtifactVersion(DATAPIPELINE_ARTIFACT_ID.getVersion()), true)
+    );
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("spark-plugins", "1.0.0"), parents,
+                      NGramTransform.class, TwitterStreamingSource.class);
```

Comment:
Not sure if I understand this. If we remove the `TwitterStreamingSource.class` entry, does it fail? if so whats the cause of failure?